Springer Books.js

```diff
@@ -0,0 +1,293 @@
+		case "book":
+			// test if any relevant <meta> information is available
+			if(ZU.xpathText(doc, '//meta[@property="og:title"]/@content')) return "book";
+			//test for relevant search entries
+		case "search": return getSearchResults(doc, true) ? "multiple" : false;
+	}
+}
+
+function doWeb(doc, url) {
+	var detection_type = detectWeb(doc, url);
+	switch (detection_type) {
+		case "book": scrapeBook(doc, url);
+		case "multiple": {
+			if (!getSearchResults(doc, true)) return;
+			Zotero.selectItems(getSearchResults(doc, false), function (items) {
+				if (!items) return true;
+				var articles = [];
+				for (var i in items) {
+					articles.push(i);
+				}
+				ZU.processDocuments(articles, scrapeBook);
+			});
+		}
+	}
+	
+}
+
+//returns true if origStr is a shorter version of newStr
+function shortenedStringTest(origStr, newStr) {
+	if (origStr.length > newStr.length) return false;
+	if (origStr.length < 10)
+		return (origStr == newStr.substring(0, origStr.length));
+	else
+		return (origStr.substring(0,9) == newStr.substring(0,9));
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var search_entries = ZU.xpath(doc, '//div[@id="result-list"]/div');
+	for (var i=0; i<search_entries.length; i++) {
+		var item_type = search_entries[i].getAttribute("class");
+		//test if we actually have a usable result
+		if (!item_type || item_type.search(/result-item result-item-\d+ result-type-book.*/) == -1) continue;
+		var title_link = ZU.xpath(search_entries[i], "h4/a");
+		if (title_link.length == 0) continue;
+		//extract book title and URL
+		var title = title_link[0].textContent;
+		var href = title_link[0].getAttribute("href");
+		if (!title || !href) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function scrapeBook(doc, url) {
+	//Call the embedded metadata translator
+	var translator = Zotero.loadTranslator("web");
+	translator.setTranslator("951c027d-74ac-47d4-a107-9c3069ab7b48");
+	translator.setDocument(doc);
+	translator.setHandler("itemDone", function(obj, item) {
+		//extract information from the title field:
+		//item.title = <book title> | <first creator> | Springer
+		var title_entries = item.title.split(/\s\|\s/);
+		item.title = title_entries[0];
+		//forward compatibility: check if the embedded metadata translator
+		//has found a creator (i.e. relevant metadata)
+		if (item.creators.length > 0) {
+			item.complete; return;
+		}
+		Z.debug("no creators found by Embedded Metadata translator");
+		//Split title into title and subtitle, verify both from
+		//a different field and complete the subtitle if neccessary;
+		if (item.title.indexOf(" - ") != -1) {
+			//Step 1: decompose item.title into the "title" field from the bibliography
+			//and the (potentially abbreviated) subtitle
+			var titlesXpath = ZU.xpath(doc, '//div[@class="product-bibliographic"]/dl/dd/div/div/dl/dd');
+			var titleCand = null;
```

Comment:
I don't think this title e.g. `Theoretical Physics 7 - Quantum Mechanics - Methods and | Wolfgang Nolting | Springer` is a good text to extract any information from. How about the information at the bottom, e.g.

![springer-bibliographic](https://user-images.githubusercontent.com/5199995/34542396-ac58eaa4-f0dc-11e7-913a-13237364df06.jpg)

? One should also check for the remaining lines here...

Reply:
The problem is that the structure of this table is not consistent across different books, and only the author, publisher, and ISBN can be accessed in a stable way. I decided to first parse the information from the metadata because that is guaranteed to be correct, but might be abbreviated. In a second step I search the information at the bottom for the complete title, subtitle, and author and check these against the data I have from step 1. That is where the function `shortenedStringTest` from above is used. I found that this is the most stable thing one can do.